common/model/bundle.c

```diff
@@ -236,6 +257,73 @@ void bundle_reset_indexes(bundle_transactions_t *const bundle) {
   }
 }
 
+void bundle_set_messages(bundle_transactions_t *bundle, signature_fragments_t *messages) {
+  iota_transaction_t *tx = NULL;
+  size_t index = 0;
+  tryte_t **msg = NULL;
+  size_t msg_len = 0;
+  tryte_t trytes_buff[NUM_TRYTES_SIGNATURE];
+
+  BUNDLE_FOREACH(bundle, tx) {
+    msg = signature_fragments_at(messages, index);
+    if (msg == NULL) {
+      break;
+    }
+    msg_len = strlen((char *)*msg);
+    memcpy(trytes_buff, *msg, msg_len);
+    memset(trytes_buff + msg_len, '9', NUM_TRYTES_SIGNATURE - msg_len);
+
+    // trytes to flex_trits
+    flex_trits_from_trytes(tx->data.signature_or_message, NUM_TRITS_SIGNATURE, trytes_buff, NUM_TRYTES_SIGNATURE,
+                           NUM_TRYTES_SIGNATURE);
```

Comment:
don't get your point, here `trytes_buff` is for padding and conversion.

Reply:
I meant that you can achieve exactly the same without the `trytes_buff `

Reply:
No, `msg` is trytes with a random length, unable to do the in-place conversion. that's why need a `trytes_buff` which length is NUM_TRYTES_SIGNATURE for padding and trytes->flex trits conversion..

Reply:
I think something like this would work but ok, I'll approve.
```
memset(tx->data.signature_or_message, FLEX_TRIT_NULL_VALUE, NUM_FLEX_TRITS_SIGNATURE);
flex_trits_from_trytes(tx->data.signature_or_message, NUM_TRITS_SIGNATURE, (tryte_t *)msg, NUM_TRYTES_SIGNATURE, msg_len);
```

Reply:
yes, it will work, if `signature_or_message` contains FLEX_TRIT_NULLs or transactions in a bundle are properly init, we should make sure it. maybe I think too much.

Reply:
No worries, I'm just writing ideas that come to my mind, they are not all good or too keep :)

Reply:
let's make sure transactions are initialized before putting in a bundle.